zebrad/src/components/metrics.rs

```diff
@@ -21,8 +21,16 @@ impl MetricsEndpoint {
     pub fn init_tokio(&mut self, tokio_component: &TokioComponent) -> Result<(), FrameworkError> {
         info!("Initializing metrics endpoint");
 
-        // XXX load metrics addr from config
-        let addr = "0.0.0.0:9999"
+        // TODO:
+        //  - load metrics addr from config
+        //  - security: disable service by default
+        //    Binding to localhost by default is insecure, because other
+        //    applications on the machine (such as a web browser) can access
+        //    the endpoint.
```

Comment:
Hmm, isn't the goal of providing this service to allow other applications (the prometheus collector) to access the endpoint?  Is the issue here that the service is enabled by default or that it's bound to the wrong address?

Reply:
The security issues are:
- the service is bound to all addresses by default, rather than localhost, and
- the service is enabled by default.

The secure default is off, and binding to localhost when turned on.

Reply:
When users turn the service on, they can make sure that it's configured securely, in the context of their specific deployment. That might mean firewalls, restricting which applications can run on the host, or just accepting the risk.

Reply:
Would it help to say "applications other than the prometheus collector", or "turning the service on by default is insecure" ?

Reply:
Ah, that makes sense, I forgot about the distinction between "bound to all addresses" and "bound to localhost". What do you think of making the config entry an `Option<SocketAddr>` and setting the default value to `None`?  Then we would have a safe default (no metrics service) and users could enable it by setting an address that was appropriate for them.

Reply:
I think that's a great idea.

Using "localhost" as the default isn't the perfect solution, because:
* it doesn't convert to SocketAddr, even though the docs say it should, and
* there are actually many localhost addresses: 127.0.0.0/8 and ::1

Using "None" means that the user can decide between IPv4 or IPv6 as needed, and we don't risk picking the wrong address. We can encourage users to specify 127.0.0.1 by providing that as an example in the default config's comments.